File: uikit/src/main/java/io/agora/chat/uikit/conversation/interfaces/IConversationListLayout.java
package io.agora.chat.uikit.conversation.interfaces;


import io.agora.chat.uikit.conversation.adapter.EaseConversationListAdapter;
import io.agora.chat.uikit.conversation.model.EaseConversationInfo;
import io.agora.chat.uikit.conversation.presenter.EaseConversationPresenter;
import io.agora.chat.uikit.interfaces.IRecyclerView;

public interface IConversationListLayout extends IRecyclerView {

    /**
     * 添加其他类型的代理类
     * @param delegate
     */
    //void addDelegate(EaseBaseConversationDelegate delegate);

    /**
     * 设置presenter
     * @param presenter
     */
    void setPresenter(EaseConversationPresenter presenter);

    /**
     * 是否展示默认的条目菜单
     * @param showDefault
     */
    void showItemDefaultMenu(boolean showDefault);

    /**
     * Set custom list adapter
     * @param listAdapter
     */
    void setListAdapter(EaseConversationListAdapter listAdapter);

    /**
     * 获取数据适配器
     * @return
     */
    EaseConversationListAdapter getListAdapter();

    /**
     * 获取条目数据
     * @param position
     * @return
     */
    EaseConversationInfo getItem(int position);


    /**
     * 将对话置为已读
     * @param position
     * @param info
     */
    void makeConversionRead(int position, EaseConversationInfo info);

    /**
     * 置顶
     * @param position
     * @param info
     */
    void makeConversationTop(int position, EaseConversationInfo info);

    /**
     * 取消置顶
     * @param position
     * @param info
     */
    void cancelConversationTop(int position, EaseConversationInfo info);

    /**
     * 删除会话
     * @param position
     * @param info
     */
    void deleteConversation(int position, EaseConversationInfo info);

    /**
     * 设置会话变化的监听
     * @param listener
     */
    void setOnConversationChangeListener(OnConversationChangeListener listener);

    /**
     * 设置加载会话状态监听
     * @param loadListener
     */
    void setOnConversationLoadListener(OnConversationLoadListener loadListener);
}

File: ease-im-kit/src/main/kotlin/io/agora/chat/uikit/feature/conversation/interfaces/IConversationListLayout.kt
package io.agora.chat.uikit.feature.conversation.interfaces

import io.agora.chat.uikit.feature.conversation.adapter.ChatUIKitConversationListAdapter
import io.agora.chat.uikit.common.interfaces.IRecyclerView
import io.agora.chat.uikit.model.ChatUIKitConversation
import io.agora.chat.uikit.viewmodel.conversations.IConversationListRequest

/**
 * Defines the conversation interface called by developers.
 */
interface IConversationListLayout: IRecyclerView {

    fun setViewModel(viewModel: IConversationListRequest?)

    /**
     * Set custom list adapter
     * @param listAdapter
     */
    fun setListAdapter(listAdapter: ChatUIKitConversationListAdapter?)

    /**
     * Get data adapter
     * @return
     */
    fun getListAdapter(): ChatUIKitConversationListAdapter?

    /**
     * Get item data
     * @param position
     * @return
     */
    fun getItem(position: Int): ChatUIKitConversation?


    /**
     * Make conversation read
     * @param position
     * @param info
     */
    fun makeConversionRead(position: Int, info: ChatUIKitConversation?)

    fun makeConversationTop(position: Int, info: ChatUIKitConversation?)

    fun cancelConversationTop(position: Int, info: ChatUIKitConversation?)

    /**
     * Delete conversation
     * @param position
     * @param info
     */
    fun deleteConversation(position: Int, info: ChatUIKitConversation?)

    /**
     * Set up monitoring of session changes
     * @param listener
     */
    fun setOnConversationChangeListener(listener: OnConversationListChangeListener?)

    /**
     * load conversation data listener
     * @param listener
     * @return
     */
    fun setLoadConversationListener(listener: OnLoadConversationListener)

}